### PR TITLE
Improved support for geographic plots

### PR DESCRIPTION
**Pull request by philippjfr**
Improves support for geographic plots and adds user guide.

### CODE REVIEW

**Comment by jbednar** [comment]
Looks good!

**Comment by philippjfr** [comment]
@jbednar Ready for another round of review.

**Comment by jbednar** [comment]
Tests are failing, though?

**Comment by philippjfr** [comment]
Just a flake.

File: README.md [hunk 1]
```diff
@@ -13,12 +13,13 @@ The PyData ecosystem has a number of core Python data containers that allow user
...
 
 HoloPlot provides a high-level plotting API built on HoloViews that provides a general and consistent API for plotting data in all the abovementioned formats. HoloPlot can integrate neatly with the individual libraries if an extension mechanism for the native plot APIs is offered, or it can be used as a standalone component.
 
-The API is currently under heavy development and is not yet ready to be considered stable. That said, the Pandas API that it mirrors is quite stable, so if you want to start playing with it anyway, have a look at the [Getting Started Guide](getting_started/index.html) and check out the current functionality in the [User Guide.](user_guide/index.html)
+The API is currently under heavy development and is not yet ready to be considered stable. That said, the Pandas API that it mirrors is quite stable, so if you want to start playing with it anyway, have a look at the [Getting Started Guide](https://pyviz.github.io/holoplot/getting_started/index.html) and check out the current functionality in the [User Guide.](https://pyviz.github.io/holoplot/user_guide/index.html)
```
Comment (jbednar) [inline README.md]: Are you ready to remove the disclaimer?  I think it's already removed from the website, so there's probably no point in having it just in the readme.

File: doc/homepage.ipynb [hunk 2]
```diff
@@ -13,6 +13,8 @@
     "* [Dask](http://dask.pydata.org): DataFrame, Series (distributed/out of core arrays and columnar data)\n",
     "* [Streamz](http://streamz.readthedocs.io): DataFrame(s), Series(s) (streaming columnar data)\n",
     "* [Intake](http://github.com/ContinuumIO/intake): DataSource (data catalogues)\n",
+    "* [GeoPandas](http://pandas.pydata.org): DataFrame, Series (columnar/tabular data)\n",
+    "\n",
     "\n",
     "Many of these libraries have the concept of a high-level plotting API that lets a user generate common plot types very easily. The native plotting APIs are generally built on [Matplotlib](http://matplotlib.org), which provides a solid foundation, but means that users miss out the benefits of modern, interactive plotting libraries for the web like [Bokeh](http://bokeh.pydata.org) and [HoloViews](http://holoviews.org).\n",
```
Comment (jbednar) [inline doc/homepage.ipynb]: Is "Several" or "Some" more accurate than "Many" now, given how long the list is?  Seems like Pandas and XArray do have these APIs already; are there others?  I don't think it's a majority...

File: holoplot/__init__.py [hunk 3]
```diff
@@ -459,6 +467,40 @@ def points(self, x=None, y=None, **kwds):
         """
         return self(x, y, kind='points', **kwds)
 
+    def polys(self, x=None, y=None, c=None, **kwds):
```
Comment (jbednar) [inline holoplot/__init__.py]: polys, not polygons?  In most cases the method name echoes the HoloViews object name, but not here?

File: holoplot/converter.py [hunk 4]
```diff
@@ -837,12 +862,12 @@ def contour(self, x=None, y=None, z=None, data=None, filled=False):
         opts = dict(plot=self._plot_opts, style=self._style_opts, norm=self._norm_opts)
 
         # Temporary workaround for bug in contours operation
-        if hv_version < '1.10.5':
-            qmesh = self.image(x, y, z, data)
-        else:
+        if hv_version > '1.10.4':
```
Comment (jbednar) [inline holoplot/converter.py]: You don't want to just require hv=1.10.5?

File: examples/user_guide/Geographic_Data.ipynb [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    "\n",
+    "    conda install -c pyviz geoviews\n",
+    "    \n",
+    "or if the cartopy dependency has been satisfied in some other way, GeoViews may also be installed using pip:\n",
+    "\n",
+    "    pip install geoviews\n",
+    "\n",
+    "## Usage\n",
+    "\n",
+    "To declare a geographic plot we have to supply a ``cartopy.crs.CRS`` (or coordinate reference system).  Coordinate reference systems are described in the [GeoViews documentation](http://geoviews.org/user_guide/Projections.html) and the full list of available CRSs is in the [cartopy documentation](https://scitools.org.uk/cartopy/docs/v0.15/crs/projections.html). Only certain HoloPlot types support geographic coordinates, currently including: 'points', 'image', 'quadmesh', 'contour' and 'contourf'.\n",
```
Comment (jbednar) [inline examples/user_guide/Geographic_Data.ipynb]: And poly (polygon) and paths?

**Comment by jbednar** [comment]
Shall I merge it?

File: holoplot/converter.py (hunk 4 above)
Comment (philippjfr) [reply holoplot/converter.py]: True, put this in before that was released.

File: examples/user_guide/Geographic_Data.ipynb (hunk 5 above)
Comment (philippjfr) [reply examples/user_guide/Geographic_Data.ipynb]: True, should add those.

File: holoplot/__init__.py (hunk 3 above)
Comment (philippjfr) [reply holoplot/__init__.py]: Suppose I should just make it ``polygons``.

File: doc/homepage.ipynb (hunk 2 above)
Comment (philippjfr) [reply doc/homepage.ipynb]: Sure, xarray, pandas and geopandas do.

File: README.md (hunk 1 above)
Comment (philippjfr) [reply README.md]: Agreed.

File: holoplot/__init__.py (hunk 3 above)
Comment (jbednar) [reply holoplot/__init__.py]: That would be my vote, but it's your call.

File: doc/homepage.ipynb (hunk 2 above)
Comment (jbednar) [reply doc/homepage.ipynb]: Ok, sounds like "Several" to me.

**Comment by philippjfr** [comment]
Once tests pass.

File: examples/user_guide/Geographic_Data.ipynb [hunk 6]
```diff
@@ -92,7 +92,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The GeoPanas support allows plotting ``GeoDataFrames`` containing ``'Point'``, ``'Polygon'``, ``'LineString'`` and ``'LineRing'`` geometries, but not ones containing a mixture of different geometry types. Calling ``.holoplot`` will automatically figure out the geometry type to plot, but it also possible to call ``.holoplot.points``, ``.holoplot.polygons``, and ``.holoplot.paths`` explicitly.\n",
+    "The GeoPandas support allows plotting ``GeoDataFrames`` containing ``'Point'``, ``'Polygon'``, ``'LineString'`` and ``'LineRing'`` geometries, but not ones containing a mixture of different geometry types. Calling ``.holoplot`` will automatically figure out the geometry type to plot, but it also possible to call ``.holoplot.points``, ``.holoplot.polygons``, and ``.holoplot.paths`` explicitly.\n",
```
Comment (philippjfr) [inline examples/user_guide/Geographic_Data.ipynb]: Ugh, thanks.

**Comment by philippjfr** [comment]
Got some minor edits which won't affect the tests. So I'll wait for it to pass, push the changes and then merge myself.

**Comment by philippjfr** [comment]
Looks like it's passing, pushing my changes and merging.

**Pull request merged**